modules/checklists/checklists.php

```diff
@@ -640,6 +640,23 @@ public function display_meta_box($post)
             // Apply filters to the list of requirements
             $requirements = apply_filters('publishpress_checklists_requirement_list', $requirements, $post);
 
+            $options = (array)get_option('publishpress_checklists_checklists_options');
+
+            $requirement_rule_array = [];
+            $new_requirements_array = [];
+            $index = 0;
+            foreach ($requirements as $requirement_key => $p_requirements ) {
+                $requirement_rule_array[$requirement_key . '_rule'] = $requirement_key;
+            }
+
+            $new_arr = array_intersect_key( $options, $requirement_rule_array );
+
+            $requirement_rule_array = array_merge(array_flip(array_keys( $new_arr) ), $requirement_rule_array);
+
+            foreach ($requirement_rule_array as $req_index) {
+                $new_requirements_array[$req_index] = $requirements[$req_index];
+            };
+
```

Comment:
I see a very similar code added to function `options_page_controller`, same file lines 877 to 893.

Please, avoid duplicated code. If you need the same code in multiple places it is required to create a new method encapsulating it, so you can call it on both places.

Please, could you refactor it to remove the duplication?

Reply:
But the feature works well now 👍  I see the sorted list both on the settings and post edit page.

Reply:
@andergmartins  I removed the duplicated codes and pushed again.
Please check commits again.

Thanks.

Reply:
Thanks @julianactrl I confirm that works.